### PR TITLE
conftest: optimize _getconftestmodules

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
Not sure about the optimization (came up with this while debugging it).
It changes the behavior to only cache directories now.

File: src/_pytest/config/__init__.py [hunk 1]
```diff
@@ -378,25 +378,27 @@ def _try_load_conftest(self, anchor):
     def _getconftestmodules(self, path):
         if self._noconftest:
             return []
+
+        if path.isfile():
```
Comment (RonnyPfannschmidt) [inline src/_pytest/config/__init__.py]: just use if/else to set directory instead of recursing

Comment (blueyed) [reply src/_pytest/config/__init__.py]: Yeah.
Do you agree with the change in general though? (to only cache dirs)

Comment (blueyed) [reply src/_pytest/config/__init__.py]: Cool, amended.

**Pull request merged**